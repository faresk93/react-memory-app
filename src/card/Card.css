.memory > .card {
  font-size: 2em;
  flex: 1 1 calc(100% / 4 - 0.4em);
  outline: 0.08em solid transparent;
  margin: 0.2em;
  display: flex;
  cursor: default;
}

.memory > .card.easy {
  flex: 1 1 calc(100% / 2 - 0.4em);
}

.memory > .card.medium {
  flex: 1 1 calc(100% / 4 - 0.4em);
}

.memory > .card.hard {
  flex: 1 1 calc(100% / 6 - 0.4em);
}

.memory > .card.hidden {
  /*background: #FFF;*/
}

.justMatched,
.justMismatched {
  outline: 0.1em solid green;
}
.justMismatched {
  outline-color: red;
}


.memory > .card.visible {
  cursor: not-allowed;
}

.memory > .card > .symbol {
  margin: auto;
}

.flip-card {
  margin: 5px;
  font-size: 2em;
  background-color: transparent;
  min-width: 50px;
  min-height: 70px;
  cursor: pointer;
  /*perspective: 1000px;*/
}

.flip-card-inner {
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: transform 0.6s;
  transform-style: preserve-3d;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
}

.flipped.flip-card-inner {
  transform: rotateY(180deg);
}

.flip-card-front, .flip-card-back {
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
}

.flip-card-front {
  background-color: #FFF;
  opacity: .8;
  color: black;
}

.flip-card-back {
  background-color: rgba(41, 128, 185, 0.51);
  color: white;
  transform: rotateY(180deg);
}

.flip-card.easy {
  flex: 1 1 calc(100% / 2 - 0.4em);
}

.flip-card.medium {
  flex: 1 1 calc(100% / 4 - 0.4em);
}

.flip-card.hard {
  flex: 1 1 calc(100% / 6 - 0.4em);
}
