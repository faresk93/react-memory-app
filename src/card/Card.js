import React from 'react'
import './Card.css'
import PropTypes from 'prop-types'

const HIDDEN_SYMBOL = '❓';
const getDifficultyClass = (difficulty) => {
    switch (difficulty) {
        case 2:
            return 'easy';
        case 4:
            return 'medium';
        case 6:
            return 'hard';
        default:
            return 'medium'
    }
};
const Card = ({card, index, feedback, difficulty, onClick}) => {
    const difficultyClass = getDifficultyClass(difficulty);
    return (
        <div className={`card ${feedback} ${difficultyClass}`}
             onClick={_ => onClick(index)}>
            <span className="symbol">
                {feedback === 'hidden' ? HIDDEN_SYMBOL : card}
            </span>
        </div>
    )
};

Card.propTypes = {
    card: PropTypes.string.isRequired,
    index: PropTypes.number.isRequired,
    feedback: PropTypes.oneOf([
        'visible', 'hidden', 'justMatched', 'justMismatched'
    ]),
    difficulty: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired

};

export default Card
