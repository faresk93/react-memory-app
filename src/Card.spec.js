import React from 'react'
import {shallow} from 'enzyme'
import {expect} from 'chai'
import sinon from 'sinon'
import Card from './card/Card'

describe('<Card/>', () => {
    it('should trigger its `onClick` prop when clicked', () => {
        // function (spy) to save all calls
        // Jest const onClick = jest.fn();
        // SINON
        const onClick = sinon.spy();
        const wrapper = shallow(<Card card="ðŸ˜Š" difficulty={4} index={0} onClick={onClick}/>)

        wrapper.simulate('click');
        expect(onClick).to.have.been.calledWith(0)
    })

    it('should match its reference snapshot', () => {
        const onClick = sinon.spy();
        const wrapper = shallow(<Card card="ðŸ˜Š" difficulty={4} index={0} onClick={onClick}/>)

        expect(wrapper).to.matchSnapshot()
    })
})
