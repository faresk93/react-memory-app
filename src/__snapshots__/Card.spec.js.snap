// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should match its reference snapshot 1`] = `
<div
  className="card undefined medium"
  onClick={[Function]}
>
  <span
    className="symbol"
  >
    ðŸ˜Š
  </span>
</div>
`;
